### PR TITLE
creates a database we can connect to and get data from

### PR DESCRIPTION
**Pull request by HilbertSpitzer**
related to  #33

### CODE REVIEW

File: app/database/db_build.sql [hunk 1]
```diff
@@ -11,6 +11,9 @@ CREATE TABLE nuggets(
   description TEXT,
   img_url VARCHAR(100),
   author VARCHAR(30) NOT NULL
-)
+);
 
--- COMMIT;
+INSERT INTO nuggets (lat,long,category,title,description,img_url,author)
+VALUES (100,100,'food','bigwave','tsunami',null,'me');
```
Comment (mattlub) [inline app/database/db_build.sql]: should have spaces after commas.

also should put it more realistic values for lat and long, e.g. -30.234903

File: app/src/server.js [hunk 2]
```diff
@@ -1,13 +1,24 @@
-const path = require('path')
+// const path = require('path')
 const express = require('express')
+const query = require('./queries.js')
 const app = express()
 
-const staticFilesPath = path.join(__dirname, '../public')
-app.use(express.static(staticFilesPath))
+// const staticFilesPath = path.join(__dirname, '../public')
+// app.use(express.static(staticFilesPath))
```
Comment (mattlub) [inline app/src/server.js]: uncomment

File: app/src/queries.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const dbConnection = require('../database/db_connection.js')
+
+const getAll = (cb) => {
+  dbConnection.query('SELECT * FROM nuggets', cb)
```
Comment (mattlub) [inline app/src/queries.js]: don't we want the function to return json or something

**Review by mattlub — CHANGES_REQUESTED** [review]

File: app/database/db_build.sql [hunk 4]
```diff
@@ -11,4 +11,10 @@ CREATE TABLE nuggets(
   description TEXT,
   img_url VARCHAR(100),
   author VARCHAR(30) NOT NULL
-)
+);
+
+INSERT INTO nuggets (lat,long,category,title,description,img_url,author)
```
Comment (mattlub) [inline app/database/db_build.sql]: spaces after commas here.

File: app/src/server.js [hunk 5]
```diff
@@ -1,15 +1,22 @@
...
-// app.get('/', function(req,res){
-//   res.send('MARIOOOOOOOO pshu pshu pshu')
-// })
+app.get('/all-nuggets', function (req, res) {
+  query.getAll((err, result) => {
+    if (err) {
+      console.log(err)
+      return
+    }
+    console.log('result is ', JSON.stringify(result.rows))
```
Comment (mattlub) [inline app/src/server.js]: this isn't very useful, we have to send a response.

File: app/src/server.js [hunk 6]
```diff
@@ -1,15 +1,22 @@
...
 app.use(express.static(staticFilesPath))
 
-// app.get('/', function(req,res){
-//   res.send('MARIOOOOOOOO pshu pshu pshu')
-// })
+app.get('/all-nuggets', function (req, res) {
+  query.getAll((err, result) => {
+    if (err) {
+      console.log(err)
+      return
```
Comment (mattlub) [inline app/src/server.js]: this request will hang, we need to send a response

File: app/src/queries.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+const dbConnection = require('../database/db_connection.js')
+
+const getAll = (cb) => {
+  dbConnection.query('SELECT * FROM nuggets', (error, result) => {
+    if (error) {
+      return cb(error)
+    }
+    return cb(null, result)
```
Comment (mattlub) [inline app/src/queries.js:8]: I think the db query function should return json. don't you?

**Review by mattlub — CHANGES_REQUESTED** [review]

File: app/src/server.js (hunk 6 above)
Comment (Karyum) [inline app/src/server.js]: yeah we will start by responding with a text then change it

File: app/src/server.js (hunk 5 above)
Comment (Karyum) [inline app/src/server.js]: yeah we will start by responding with a text then change it

File: app/src/queries.js (hunk 7 above)
Comment (Karyum) [reply app/src/queries.js:8]: i disagree doing it in the db query im gonna do it using express from the server

File: app/src/server.js [hunk 8]
```diff
@@ -1,15 +1,22 @@
...
 app.use(express.static(staticFilesPath))
 
-// app.get('/', function(req,res){
-//   res.send('MARIOOOOOOOO pshu pshu pshu')
-// })
+app.get('/all-nuggets', function (req, res) {
+  query.getAll((err, result) => {
+    if (err) {
+      console.log(err)
+      res.send('DATABSE ERROR')
```
Comment (mattlub) [inline app/src/server.js]: I think you need a return inside this block, otherwise there will be an error (maybe)
i.e. return res.send(...

**Review by mattlub — CHANGES_REQUESTED** [review]

File: app/src/server.js (hunk 8 above)
Comment (Karyum) [reply app/src/server.js]: we tested this and the res.send works as a return

**Review by mattlub — APPROVED** [review]

**Pull request merged**